paddle/fluid/operators/mkldnn/sum_mkldnn_op.cc

```diff
@@ -45,82 +45,63 @@ using paddle::platform::MKLDNNDeviceContext;
 
-      : platform::MKLDNNHandlerT<T, dnnl::sum>(
-            dev_ctx, dev_ctx.GetEngine(), cpu_place,
-            platform::CreateKey(dev_ctx, framework::vectorize(z->dims()),
-                                uniq_name)),
+      : platform::MKLDNNHandlerNoCachingT<T, dnnl::sum>(engine, cpu_place),
         num_inputs_(0) {
-    for (size_t i = 0; i < in_vars.size(); i++) {
-      srcs_suffix_.push_back(std::string("-") + std::to_string(i));
-    }
+    auto dst_tz = framework::vectorize<int64_t>(z->dims());
+    auto src_tz = dst_tz;
 
-    if (!this->isCached()) {
-      auto dst_tz = framework::vectorize<int64_t>(z->dims());
-      auto src_tz = dst_tz;
-
-      std::vector<mkldnn::memory::desc> srcs_md;
-      for (size_t i = 0; i < in_vars.size(); i++) {
-        auto& input_it = in_vars[i]->Get<framework::LoDTensor>();
-        if (input_it.numel() == 0) {
-          continue;
-        }
-        MKLDNNMemoryFormat input_format = input_it.format();
-        srcs_md.push_back(mkldnn::memory::desc(
-            src_tz, platform::MKLDNNGetDataType<T>(), input_format));
-        ++num_inputs_;
+    std::vector<mkldnn::memory::desc> srcs_md;
+    for (size_t i = 0; i < in_vars.size(); i++) {
+      auto& input_it = in_vars[i]->Get<framework::LoDTensor>();
+      if (input_it.numel() == 0) {
+        continue;
       }
-      std::vector<float> scales(num_inputs_, 1.0);
+      MKLDNNMemoryFormat input_format = input_it.format();
+      srcs_md.push_back(mkldnn::memory::desc(
+          src_tz, platform::MKLDNNGetDataType<T>(), input_format));
+      ++num_inputs_;
+    }
+    std::vector<float> scales(num_inputs_, 1.0);
 
-      auto dst_md = mkldnn::memory::desc(
-          dst_tz, platform::MKLDNNGetDataType<T>(), MKLDNNMemoryFormat::any);
+    auto dst_md = mkldnn::memory::desc(dst_tz, platform::MKLDNNGetDataType<T>(),
+                                       MKLDNNMemoryFormat::any);
 
-      this->AcquireForwardPrimitiveDescriptor(dst_md, scales, srcs_md);
-    }
+    this->AcquireForwardPrimitiveDescriptor(dst_md, scales, srcs_md);
   }
 
   // (jczaja) sum oneDNN prim is not having .desc attribute so
   // we cannot use base AcquireForwardPrimitiveDescriptor
   void AcquireForwardPrimitiveDescriptor(
       const mkldnn::memory::desc& dst_md, const std::vector<float>& scales,
       const std::vector<mkldnn::memory::desc>& srcs_md) {
-    // Sum op does not have backward so no passing from FWD to BWD is needed
-    const std::string key_pd = this->key_ + "@fwd_pd";
-    this->fwd_pd_ = std::static_pointer_cast<dnnl::sum::primitive_desc>(
-        this->dev_ctx_.GetBlob(key_pd));
-    if (this->fwd_pd_ == nullptr) {
-      this->fwd_pd_.reset(new dnnl::sum::primitive_desc(dst_md, scales, srcs_md,
-                                                        this->engine_));
-      this->dev_ctx_.SetBlob(key_pd, this->fwd_pd_);
-    }
+    this->fwd_pd_.reset(
+        new dnnl::sum::primitive_desc(dst_md, scales, srcs_md, this->engine_));
```

Comment:
You are using "new" here. Is it later destroyed anywhere?

Reply:
this->fwd_pd_ is a shared pointer so when reference count will get to zero then deallocation of  of object will happen . So we do not have to worry about memory leaks.

Reply:
Ok, thanks. I didn't see that it is a smart pointer